File: php/features.php
<?php
declare(strict_types=1);

// Hackernews SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class HackernewsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new HackernewsBaseFeature();
            case "test":
                return new HackernewsTestFeature();
            default:
                return new HackernewsBaseFeature();
        }
    }
}

File: py/features.py
# Hackernews SDK feature factory

from feature.base_feature import HackernewsBaseFeature
from feature.test_feature import HackernewsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: HackernewsBaseFeature(),
        "test": lambda: HackernewsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
